### PR TITLE
SUS-1276 | sunset WikiaHubsApiController

### PR DESCRIPTION
**Pull request by macbre**
https://wikia-inc.atlassian.net/browse/SUS-1276

Break cross-extension dependency between `WikiaHomePage` and `WikiaHubsServices`.

API tests removed in https://github.com/Wikia/api-tests/pull/275

### CODE REVIEW

File: extensions/wikia/WikiaHomePage/WikiaHomePageController.class.php [hunk 1]
```diff
@@ -479,33 +479,23 @@ public function getMarketingImage( $imgName ) {
...
 
 	/**
 	 * Get list of hubs v3 in selected language
 	 *
 	 * @param $langCode
-	 * @return null
+	 * @return array
+	 * @deprecated
 	 */
 	private function getHubsV3List( $langCode ) {
```
Comment (WikiaTech) [inline extensions/wikia/WikiaHomePage/WikiaHomePageController.class.php]: MAJOR Remove the unused function parameter "$langCode". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/WikiaHomePage/WikiaHomePage.setup.php [hunk 2]
```diff
@@ -44,5 +43,4 @@
 $wgHooks['ArticleCommentCheck'][] = 'WikiaHomePageController::onArticleCommentCheck';
 $wgHooks['AfterGlobalHeader'][] = 'WikiaHomePageController::onAfterGlobalHeader';
 $wgHooks['BeforePageDisplay'][] = 'WikiaHomePageController::onBeforePageDisplay';
-$wgHooks['AfterOasisSettingsInitialized'][] = 'WikiaHomePageController::onAfterOasisSettingsInitialized';
```
Comment (mszabo-wikia) [inline extensions/wikia/WikiaHomePage/WikiaHomePage.setup.php:47]: Let's kill this hook, there seems to be nothing else using it

**Review by mszabo-wikia — CHANGES_REQUESTED** [review]

**Review by mszabo-wikia — APPROVED** [review]

File: extensions/wikia/WikiaHomePage/WikiaHomePage.setup.php (hunk 2 above)
Comment (macbre) [reply extensions/wikia/WikiaHomePage/WikiaHomePage.setup.php:47]: Removed in c685ced

**Pull request merged**